.github/workflows/all-in-one-build.yml

```
@@ -0,0 +1,57 @@
+name: Build all-in-one
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  all-in-one:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: true
+
+    - name: Fetch git tags
+      run: |
+        git fetch --prune --unshallow --tags
+
+    - uses: actions/setup-go@v2
+      with:
+        go-version: ^1.15
+
+    - uses: actions/setup-node@v2-beta
+      with:
+        node-version: '10'
+
+    - uses: docker/login-action@v1
+      id: dockerhub-login
+      with:
+        username: jaegertracingbot
+        password: ${{ secrets.DOCKERHUB_TOKEN }}
+      env:
+        DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
+      if: env.DOCKERHUB_TOKEN != null
+
+    - name: Export DOCKERHUB_LOGIN variable
+      run: |
+        echo "DOCKERHUB_LOGIN=true" >> $GITHUB_ENV
+      if: steps.dockerhub-login.outcome == 'success'
+
+    - name: Export BRANCH variable for pull_request event
+      run: |
+        echo "BRANCH=${GITHUB_HEAD_REF}" >> $GITHUB_ENV
```

it would be better for debugging to print the derived value
```suggestion
        export BRANCH=${GITHUB_HEAD_REF}
        echo "we are on branch=$BRANCH"
        echo "BRANCH=${BRANCH}" >> $GITHUB_ENV
```